src/pasta/mod.rs

```diff
@@ -25,3 +29,300 @@ impl crate::CurveAffineExt for EqAffine {
         unimplemented!();
     }
 }
+
+/// Lexicographic comparison of Montgomery forms.
+#[inline(always)]
+const fn is_less_than(x: &[u64; 4], y: &[u64; 4]) -> bool {
+    let (_, borrow) = sbb(x[0], y[0], 0);
+    let (_, borrow) = sbb(x[1], y[1], borrow);
+    let (_, borrow) = sbb(x[2], y[2], borrow);
+    let (_, borrow) = sbb(x[3], y[3], borrow);
+    borrow >> 63 == 1
+}
+
+const FP_SIZE: usize = 32;
+const FQ_SIZE: usize = 32;
+
+struct AccessibleFp(pub [u64; 4]);
+
+impl From<AccessibleFp> for Fp {
+    fn from(fp: AccessibleFp) -> Fp {
+        let scalar = unsafe { std::mem::transmute::<AccessibleFp, Fp>(fp) };
+        scalar
+    }
+}
+
+impl From<Fp> for AccessibleFp {
+    fn from(scalar: Fp) -> AccessibleFp {
+        let fp = unsafe { std::mem::transmute::<Fp, AccessibleFp>(scalar) };
+        fp
+    }
+}
+
+impl SerdeObject for Fp {
+    fn from_raw_bytes_unchecked(bytes: &[u8]) -> Self {
+        debug_assert_eq!(bytes.len(), FP_SIZE);
+        let inner = [0, 8, 16, 24].map(|i| u64::from_le_bytes(bytes[i..i + 8].try_into().unwrap()));
+        Fp::from(AccessibleFp(inner))
+    }
+    fn from_raw_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != FP_SIZE {
+            return None;
+        }
+        let elt = Self::from_raw_bytes_unchecked(bytes);
+        Some(elt)
+    }
+    fn to_raw_bytes(&self) -> Vec<u8> {
+        // The transmute is the only way to get access to the internal array.
+        // Note that this is safe as long as the underlying array length is the same
+        // and the types contained are the same too.
+        let limbs: [u64; 4] = unsafe { std::mem::transmute_copy(&self) };
```

Comment:
This will force us to clone when not needed in reality. THat's why I don't like it. 
I can do it if you feel like it has to. But I'd prefer avoiding and having faster serialization.